src/Microsoft.TestPlatform.TestHostProvider/Hosting/DefaultTestHostManager.cs

```diff
@@ -365,10 +365,9 @@ private bool LaunchHost(TestProcessStartInfo testHostStartInfo, CancellationToke
             this.testHostProcessStdError = new StringBuilder(0, CoreUtilities.Constants.StandardErrorMaxLength);
             EqtTrace.Verbose("Launching default test Host Process {0} with arguments {1}", testHostStartInfo.FileName, testHostStartInfo.Arguments);
 
-            if (this.customTestHostLauncher == null)
+            if (this.customTestHostLauncher == null || this.customTestHostLauncher.IsDebug)
```

Comment:
A comment would help here. Is the intent to ensure that we only enter here if this is `ITestHostLauncher2` (because we know we will attach to the process separately later)?

Reply:
Done.